be/src/column/fixed_length_column_base.cpp

```diff
@@ -42,11 +42,37 @@ void FixedLengthColumnBase<T>::append(const Column& src, size_t offset, size_t c
 template <typename T>
 void FixedLengthColumnBase<T>::append_selective(const Column& src, const uint32_t* indexes, uint32_t from,
                                                 uint32_t size) {
+    indexes += from;
     const T* src_data = reinterpret_cast<const T*>(src.raw_data());
-    size_t orig_size = _data.size();
+
+    const size_t orig_size = _data.size();
     _data.resize(orig_size + size);
-    for (size_t i = 0; i < size; ++i) {
-        _data[orig_size + i] = src_data[indexes[from + i]];
+    auto* dest_data = _data.data() + orig_size;
+
+    static constexpr uint32_t SIMD_WIDTH = [] {
+#ifdef __AVX2__
+        return 256;
+#elif defined(__ARM_NEON) && defined(__aarch64__)
+        return 128;
+#else
+        return 256;
+#endif
+    }();
+    static constexpr uint32_t W = SIMD_WIDTH / (8 * sizeof(T));
+    T buffer[W];
+    size_t i = 0;
+    for (; i + W <= size; i += W) {
+        for (int j = 0; j < W; j++) {
+            buffer[j] = src_data[indexes[i + j]];
```

Comment:
Previously, I recognized the benefits of this optimization, which can be categorized into two aspects: first, SIMD, and second, a kind of manual "store-buffer." However, I abandoned the idea because it made the code overly complex. 
Now, I wonder if we can make this optimization more general, perhaps similar to a memcpy function? In this way it can used in many places.

Reply:
Yes. And the third advantage is that it may be benefit from hardware parallel memory access, since we manually unroll the for-loop and multiple random memory access instructions are closely.

Reply:
Make sense. Let me abstract a method to do this.

Reply:
Extract to `SIMDGather::gather`.

Reply:
great job